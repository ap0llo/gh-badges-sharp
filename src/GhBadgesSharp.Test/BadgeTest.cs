using System.Linq;
using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using Xunit;

namespace GhBadgesSharp.Test
{   
    [UseReporter(typeof(DiffReporter))]
    public partial class BadgeTest
    {
        private class ApprovalNamer : UnitTestFrameworkNamer
        {
            private readonly int m_Id;

            public override string Name => $"{base.Name}_{m_Id:000}";

            public ApprovalNamer(int id)
            {
                m_Id = id;
            }

            public override string GetSubdirectory() => "/testdata";
        }


        [Theory]
        //          id  template                leftText   rightText  color       labelColor    logo    logoPosition    logoWidth    link1                  link2
        [InlineData(1,  BadgeStyle.Flat,        "Hello",   "World",   "yellow",   null,         null,   null,           null,        null,                  null)]
        [InlineData(2,  BadgeStyle.Flat,        "Hello",   "World",   "yellow",   null,         null,   null,           null,        "http://example.com",  null)]
        [InlineData(3,  BadgeStyle.Plastic,     "Hello",   "World",   "yellow",   null,         null,   null,           null,        null,                  null)]
        [InlineData(4,  BadgeStyle.Plastic,     "Hello",   "World",   "yellow",   null,         null,   null,           null,        "http://example.com",  null)]
        [InlineData(5,  BadgeStyle.Flat,        "Hello",   "World",   "yellow",   null,         null,   null,           null,        "http://example.com",  "https://www.github.com")]
        [InlineData(6,  BadgeStyle.Plastic,     "Hello",   "World",   "yellow",   null,         null,   null,           null,        "http://example.com",  "https://www.github.com")]
        [InlineData(7,  BadgeStyle.Plastic,     "Hello",   "World",   "yellow",   null,         null,   null,           null,        "",                    null)]
        [InlineData(8,  BadgeStyle.Flat,        "Hello",   "World",   "yellow",   null,         null,   null,           null,        "",                    null)]
        [InlineData(9,  BadgeStyle.Flat,        "Hello",   "World",   "red",      "blue",       null,   null,           null,        null,                  null)]
        [InlineData(10, BadgeStyle.Plastic,     "Hello",   "World",   "red",      "blue",       null,   null,           null,        null,                  null)]
        [InlineData(11, BadgeStyle.FlatSquare,  "Hello",   "World",   "yellow",   null,         null,   null,           null,        null,                  null)]
        [InlineData(12, BadgeStyle.FlatSquare,  "Hello",   "World",   "yellow",   null,         null,   null,           null,        "http://example.com",  null)]
        [InlineData(13, BadgeStyle.FlatSquare,  "Hello",   "World",   "yellow",   null,         null,   null,           null,        "http://example.com",  "https://www.github.com")]
        [InlineData(14, BadgeStyle.FlatSquare,  "Hello",   "World",   "yellow",   null,         null,   null,           null,        "",                    null)]
        [InlineData(15, BadgeStyle.FlatSquare,  "Hello",   "World",   "red",      "blue",       null,   null,           null,        null,                  null)]
        [InlineData(16, BadgeStyle.ForTheBadge, "Hello",   "World",   "yellow",   null,         null,   null,           null,        null,                  null)]
        [InlineData(17, BadgeStyle.ForTheBadge, "Hello",   "World",   "yellow",   null,         null,   null,           null,        "http://example.com",  null)]
        [InlineData(18, BadgeStyle.ForTheBadge, "Hello",   "World",   "yellow",   null,         null,   null,           null,        "http://example.com",  "https://www.github.com")]
        [InlineData(19, BadgeStyle.ForTheBadge, "Hello",   "World",   "yellow",   null,         null,   null,           null,        "",                    null)]
        [InlineData(20, BadgeStyle.ForTheBadge, "Hello",   "World",   "red",      "blue",       null,   null,           null,        null,                  null)]
        public void MakeBadge_returns_expected_svg(int id, BadgeStyle style, string leftText, string rightText, string color, string labelColor, string logo, int? logoPosition, int? logoWidth, string link1, string link2)
        {
            var badge = Badge.MakeBadge(style, leftText, rightText, color, labelColor, logo, logoPosition, logoWidth, link1, link2);
            var writer = new ApprovalTextWriter(badge.ToString(), "svg");

            Approvals.Verify(writer, new ApprovalNamer(id), Approvals.GetReporter());            
        }   

    }
}
