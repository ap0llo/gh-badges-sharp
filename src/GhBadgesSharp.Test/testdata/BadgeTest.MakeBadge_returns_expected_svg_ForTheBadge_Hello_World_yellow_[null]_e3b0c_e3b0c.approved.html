<!DOCTYPE html>
            <html>
            <head>
                <style>
                body {
                    font-family: Arial, Helvetica, sans-serif;
                    font-size: 80%;
                }
                h1 {
                    font-size: 14pt;
                }
                thead {
                    font-weight: bold
                }
                table {
                    border-collapse: collapse;
                }
                table, th, td {
                    border: 1px solid black;
                }
                td {
                    padding: 0 10px 0 5px;
                }
                </style>
            </head>

            <body>

            <h1>Rendered Badge:</h1>
            <svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="135.407" height="28">
  <g shape-rendering="crispEdges">
    <rect width="63.626" height="28" fill="#555" />
    <rect x="63.626" width="71.781" height="28" fill="#dfb317" />
  </g>
  <g fill="#fff" text-anchor="middle" font-family="DejaVu Sans,Verdana,Geneva,sans-serif" font-size="100">
    <text x="318.13" y="175" transform="scale(0.1)" textLength="396.26" lengthAdjust="spacing">HELLO</text>
    <text x="995.165" y="175" font-weight="bold" transform="scale(0.1)" textLength="477.81" lengthAdjust="spacing">WORLD</text>
  </g>
</svg>
            <h1>Parameters:</h1>          
              <table>
                <thead>
                  <tr> <td>Name</td> <td>Value</td> </tr>
                </thead>
                <tbody>
                  <tr> <td>Style</td> <td>ForTheBadge</td> </tr>
                  <tr> <td>LeftText</td> <td>Hello</td> </tr>
                  <tr> <td>RightText</td> <td>World</td> </tr>
                  <tr> <td>Color</td> <td>yellow</td> </tr>
                  <tr> <td>LabelColor</td> <td></td> </tr>
                  <tr> <td>LeftLink</td> <td></td> </tr>
                  <tr> <td>RightLink</td> <td></td> </tr>
                </tbody>
              </table>

            </body>
            </html> 